Core/include/Acts/Vertexing/VertexingOptions.hpp

```diff
@@ -24,22 +24,44 @@ struct VertexingOptions {
   /// Default constructor is deleted
   VertexingOptions() = delete;
 
-  /// VertexingOptions with context
+  /// VertexingOptions with context and beam spot constraint
   ///
   /// @param gctx The goemetry context for this fit
   /// @param mctx The magnetic context for this fit
-  /// @param vconstr The pointing constraint to a vertex
-  VertexingOptions(
-      const GeometryContext& gctx, const MagneticFieldContext& mctx,
-      const Vertex<input_track_t>& vconstr = Vertex<input_track_t>())
-      : geoContext(gctx), magFieldContext(mctx), vertexConstraint(vconstr) {}
+  /// @param bsConstr The beamspot constraint
+  /// @param useConstr Boolean indicating whether vertex constraint should be used
+  VertexingOptions(const GeometryContext& gctx,
+                   const MagneticFieldContext& mctx,
+                   const Vertex<input_track_t>& bsConstr,
+                   const bool useConstr = true)
+      : geoContext(gctx),
+        magFieldContext(mctx),
+        beamSpotConstraint(bsConstr),
+        useBeamSpotConstraint(useConstr) {
+    assert((!useBeamSpotConstraint ||
```

Comment:
Should this be checked even when not in `DEBUG`?

Reply:
Like this?

Reply:
I was just trying to say that `assert` only executes in `DEBUG` builds. Should we check this consistency in all case to be sure?